sycl/test/basic_tests/parallel_for_range.cpp

```diff
@@ -1,93 +1,616 @@
 // RUN: %clangxx -fsycl %s -o %t.out
-// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
 
 #include <CL/sycl.hpp>
 
+#include <iostream>
+
 using namespace cl::sycl;
 
+[[cl::reqd_work_group_size(4, 4, 4)]] void reqd_wg_size_helper() {
+  // do nothing
+}
+
 int main() {
-  auto asyncHandler = [](exception_list es) {
-    for (auto& e : es) {
-      std::rethrow_exception(e);
+  auto AsyncHandler = [](exception_list ES) {
+    for (auto &E : ES) {
+      std::rethrow_exception(E);
     }
   };
 
-  // parallel_for, 100 global, 3 local -> fail.
+  queue Q(AsyncHandler);
+  device D(Q.get_device());
+
+  string_class DeviceVendorName = D.get_info<info::device::vendor>();
+  auto DeviceType = D.get_info<info::device::device_type>();
+
+  // parallel_for, (16, 16, 16) global, (8, 8, 8) local, reqd_wg_size(4, 4, 4)
+  // -> fail
   try {
-    queue q(asyncHandler);
-    q.submit([&](handler &cgh) {
-        cgh.parallel_for<class a>(nd_range<1>(range<1>(100), range<1>(3)),
-            [=](nd_item<1> id) {});
+    Q.submit([&](handler &CGH) {
+      CGH.parallel_for<class ReqdWGSizeNegativeA>(
+          nd_range<3>(range<3>(16, 16, 16), range<3>(8, 8, 8)),
+          [=](nd_item<3>) { reqd_wg_size_helper(); });
     });
-    q.wait_and_throw();
-    assert(false && "Should have thrown exception");
-  } catch (nd_range_error e) {
-    // We expect an error to be thrown!
+    Q.wait_and_throw();
+    std::cout << "Test case ReqdWGSizeNegativeA failed: no exception has been "
```

Comment:
```suggestion
    std::cerr << "Test case ReqdWGSizeNegativeA failed: no exception has been "
```
Please, apply to all prints in the tests.

Reply:
Done